r/tools/nixlibs.R

```diff
@@ -52,6 +42,24 @@ try_download <- function(from_url, to_file) {
   !inherits(status, "try-error") && status == 0
 }
 
+# For local debugging, set ARROW_R_DEV=TRUE to make this script print more
+quietly <- !env_is("ARROW_R_DEV", "true")
+
+# Default is build from source, not download a binary
+build_ok <- !env_is("LIBARROW_BUILD", "false")
+binary_ok <- !(env_is("LIBARROW_BINARY", "false") || env_is("LIBARROW_BINARY", ""))
+
+# Check if we're doing an offline build.
+# (Note that cmake will still be downloaded if necessary
+#  https://arrow.apache.org/docs/developers/cpp/building.html#offline-builds)
+download_ok <- !env_is("TEST_OFFLINE_BUILD", "true") && try_download("https://github.com", tempfile())
+
+# This path, within the tar file, might exist if
+# create_package_with_all_dependencies() was run. Otherwise, it won't, but
+# tools/cpp/thirdparty/ still will.
+thirdparty_dependency_dir <- "tools/cpp/thirdparty/download"
```

Comment:
Sure, I guess it could make updating an offline system easier. If the dependencies haven't changed, you could reuse the same downloaded files without re-running `create_package_with_all_dependencies()`. (If dependencies have changed, `cmake` will error because the checksums won't match.)

Reply:
Should `create_package_with_all_dependencies` also check `ARROW_THIRDPARTY_DEPENDENCY_DIR` before downloading?